### PR TITLE
Add falling back to fetch based pinging for onDemandEntries

### LINKED ISSUE #6296: HMR reloading page every 10 seconds
# Bug report

## Describe the bug

After upgrading to next@8, HMR keeps reloading the page every ~10 seconds. The console throws the following error: 
Connection to https://mydomain.com/_next/webpack-hmr was interrupted while the page was loading.

## To Reproduce
Upgrade a project from next@7.0.2 to next@8.0.1 using

    yarn upgrade next@latest react@latest react-dom@latest

## Expected behavior
HMR only reloading page when changes are made

## Screenshots
screenshot from 2019-02-14 14-06-50

## System information
 - OS: Debian
 - Browser: Firefox Developer Edition
 - Version of Next.js: 8.0.1

### PR DESCRIPTION
**Pull request by ijjk**
After discussion, I added falling back to fetch based pinging when the WebSocket fails to connect. I also added an example of how to proxy the onDemandEntries WebSocket when using a custom server. Fixes: #6296

### CODE REVIEW

**Review by timneutkens — APPROVED** [review]
Looks good to me!

File: packages/next/client/on-demand-entries-client.js [hunk 1]
```diff
@@ -49,11 +61,36 @@ export default async ({ assetPrefix }) => {
...
 
   async function ping () {
-    if (ws.readyState === ws.OPEN) {
-      ws.send(Router.pathname)
+    // Use WebSocket if available
+    if (ws && ws.readyState === ws.OPEN) {
+      return ws.send(Router.pathname)
+    }
+    if (!showedWarning) {
+      console.warn('onDemandEntries WebSocket failed to connect, falling back to fetch based pinging..')
```
Comment (timneutkens) [inline packages/next/client/on-demand-entries-client.js]: Let's add an err.sh link for this explaining why the fallbakc happens.

**Review by Timer — APPROVED** [review]
LGTM but I'd love to see some tests for the polling (fallback) mode in the future.

File: packages/next/client/on-demand-entries-client.js (hunk 1 above)
Comment (ijjk) [reply packages/next/client/on-demand-entries-client.js]: Sounds good, added.

**Comment by ijjk** [comment]
@Timer I added another test to `test/integration/ondemand/test/index.test.js` to check the fallback fetch

**Review by timneutkens — APPROVED** [review]

**Pull request merged**